zipline/data/data_portal.py

```diff
@@ -0,0 +1,1186 @@
+
+        """
+        if self._adjustment_reader is None:
+            return []
+
+        if sid not in adjustments_dict:
+            adjustments_for_sid = self._adjustment_reader.\
+                get_adjustments_for_sid(table_name, sid)
+            adjustments_dict[sid] = adjustments_for_sid
+
+        return adjustments_dict[sid]
+
+    def get_equity_price_view(self, asset):
+        """
+        Returns a DataPortalSidView for the given asset.  Used to support the
+        data[sid(N)] public API.  Not needed if DataPortal is used standalone.
+
+        Parameters
+        ----------
+        asset : Asset
+            Asset that is being queried.
+
+        Returns
+        -------
+        DataPortalSidView: Accessor into the given asset's data.
+        """
+        try:
+            view = self.views[asset]
+        except KeyError:
+            view = self.views[asset] = DataPortalSidView(asset, self)
+
+        return view
+
+    def _check_is_currently_alive(self, name, dt):
+        if dt is None:
+            dt = self.current_day
+
+        if name not in self._asset_start_dates:
+            self._get_asset_start_date(name)
+
+        start_date = self._asset_start_dates[name]
+        if self._asset_start_dates[name] > dt:
+            raise NoTradeDataAvailableTooEarly(
+                sid=name,
+                dt=dt,
+                start_dt=start_date
+            )
+
+        end_date = self._asset_end_dates[name]
+        if self._asset_end_dates[name] < dt:
+            raise NoTradeDataAvailableTooLate(
+                sid=name,
+                dt=dt,
+                end_dt=end_date
+            )
+
+    def _get_asset_start_date(self, sid):
+        if sid not in self._asset_start_dates:
+            asset = self._asset_finder.retrieve_asset(sid)
+            self._asset_start_dates[sid] = asset.start_date
+            self._asset_end_dates[sid] = asset.end_date
+
+        return self._asset_start_dates[sid]
+
+    def get_splits(self, sids, dt):
+        """
+        Returns any splits for the given sids and the given dt.
+
+        Parameters
+        ----------
+        sids : list
+            Sids for which we want splits.
+
+        dt: pd.Timestamp
+            The date for which we are checking for splits.  Note: this is
+            expected to be midnight UTC.
+
+        Returns
+        -------
+        list: List of splits, where each split is a (sid, ratio) tuple.
+        """
+        if self._adjustment_reader is None or len(sids) == 0:
+            return {}
+
+        # convert dt to # of seconds since epoch, because that's what we use
+        # in the adjustments db
+        seconds = int(dt.value / 1e9)
+
+        splits = self._adjustment_reader.conn.execute(
+            "SELECT sid, ratio FROM SPLITS WHERE effective_date = ?",
+            (seconds,)).fetchall()
+
+        sids_set = set(sids)
+        splits = [split for split in splits if split[0] in sids_set]
```

Comment:
Here it's less of a concern because we see the query above, but elsewhere code has to know that `split` is actually a tuple of sid and ratio.  I know sqlalchemy provides a namedtuple-like interface for these rows - if we're not using an ORM (too slow?), could we wrap these rows ourselves or have consumers unpack the tuples so that it's clearer what `split[0]` means?  (I had seen it in other code and had to find this method to get the answer.)